### PR TITLE
added windows support

### PR DESCRIPTION
**Pull request by xTarzx**
udp.cpp
  - use winsock api
CMakeLists.txt
  - add windows specific includes
 
removed "using namespace std" from all files to avoid "byte" name clash

build script for windows included

### CODE REVIEW

File: include/bulb.h [hunk 1]
```diff
@@ -4,18 +4,22 @@
...
+
 class Bulb
 {
 
 public:
     Bulb();
     ~Bulb();
 
-    void setDeviceIP(const std::string& ip);
+    void setDeviceIP(const std::string &ip);
```
Comment (srisham) [inline include/bulb.h:18]: Most developers would prefer writing `std::string& ` than `std::string &`

File: src/bulb.cpp [hunk 2]
```diff
@@ -250,34 +250,39 @@ std::string Bulb::setScene(ushort scene)
...
-std::string Bulb::parseResponse(std::string jsonStr, std::string addlParams) {
+std::string Bulb::parseResponse(std::string jsonStr, std::string addlParams)
+{
 
     if (jsonStr.empty())
         return jsonStr;
-    
-    json_t* root = json_object();
+
+    json_t *root = json_object();
```
Comment (srisham) [inline src/bulb.cpp:259]: Same here. Can you please undo these changes? I would prefer `<type>* ` than `<type> *`

**Review by srisham — CHANGES_REQUESTED** [review]
1. Most developers would prefer writing `<type>* `  than `<type> *`. Please revert it.
2. And squash your 2 commits into 1 and update the PR.

**Review by xTarzx — COMMENTED** [review]
Changes done.

**Review by xTarzx — COMMENTED** [review]
not sure how this works

**Review by srisham — APPROVED** [review]

**Pull request merged**